packages/@aws-cdk/aws-amplify-alpha/lib/domain.ts

```diff
@@ -131,6 +131,13 @@ export class Domain extends Resource {
     this.subDomains = props.subDomains || [];
 
     const domainName = props.domainName || id;
+    if (domainName.length > 255) {
```

Comment:
I think we might need to check token.

```typescript
Token.isUnresolved(domainName)
```

Reply:
I keep forgetting to check the token no matter how many times I try... Thank you for the reminder!